backend/checkmach.ml

```diff
@@ -77,6 +77,21 @@ module Witness = struct
     Format.fprintf ppf "%a {%a}@," print_kind kind Debuginfo.print_compact dbg
 end
 
+(* CR gyorsh: avoid duplicating somehwat tricky logic. is there a nicer way
```

Comment:
Maybe the sequence is the good abstraction,
there is `Seq.take` to the function would only
need to take `to_seq` and `of_seq` to provide
the same functionality.

Reply:
good idea! we still need `cardinal` to avoid reallocating on inputs shorter than `n` (I'm not sure Seq is smart enough to avoid it).